src/data-grid/data-grid.tsx

```diff
@@ -692,18 +700,31 @@ const DataGrid: React.FunctionComponent<Props> = p => {
         ctx.rect(0, headerHeight + 1, width, height - headerHeight - 1);
         ctx.clip();
         ctx.beginPath();
-        // draw cell contents
-        let row = cellYOffset;
-        {
-            // we are going to set the font out here because setting this in FF is slow
-            // for no obvious reason. It will get saved/restored as needed anyway. All cells
-            // can depend on the default font being set and not need to set it themselves.
-            ctx.font = "13px Roboto, sans-serif";
+
+        let x = 0;
+        let clipX = 0;
+        let row = 0;
+        ctx.font = "13px Roboto, sans-serif";
+        for (const c of effectiveCols) {
             let y = headerHeight + translateY;
+            row = cellYOffset;
+
+            ctx.save();
+            ctx.beginPath();
+            if (c.sticky) {
+                clipX = Math.max(clipX, x + c.width);
```

Comment:
nitpick but I don't see x change anywhere, and it's set to 0 on 704.

Reply:
god catch!

Reply:
Actually line 821 it gets set

Reply:
mb, thought 821 was referring to (deleted) 704's x